### ISSUE TITLE
Problems when feeding a list of subreddits

### ISSUE THREAD

**Issue by nihelmasell** [aliparlakci/bulk-downloader-for-reddit#52]
Hi, I don't know if this helps, but I am having some issues when I try to feed a list of subreddits via cat to your software. I tried it with your last version, but I couldn't fully run the "script", as the "press enter to quit" message makes the script stop (I don't know how to send an "enter" into a command line script"). I had to go back to your jul 22 version, which didn't have this issue, and allowed me to download hundreds of subreddits unattended (mostly). Great script anyway!

**Comment by aliparlakci** [comment]
Can you explain, step-by-step, how do you end up on "Press enter quit..."?
And provide your CONSOLE_LOG.txt file if possible? It is located inside
LOG_FILES in the directory you give.

**Comment by nihelmasell** [comment]
> Can you explain, step-by-step, how do you end up on "Press enter quit..."? And provide your CONSOLE_LOG.txt file if possible? It is located inside LOG_FILES in the directory you give.

The command I'm using is 
cat bulkdownloaderforredditlist.txt | xargs -I {} python3 /Volumes/WD-8TB-NOV2018/bulk-downloader-for-reddit-master/script.py --directory /Volumes/WD-8TB-NOV2018/bulk-downloader-for-reddit-master/Backup --sort hot --subreddit {}

I've been using the July´s version for a week, so I´ll have to perform a thorough search for the log of your latest version. I´ll try to send it tomorrow.

**Comment by aliparlakci** [comment]
First of all I strongly recommend you to use the latest version of the
script and report the bug you encounter. I usually can fix fix it within a
day.

For multi subreddits arguments should be following: --subreddits
gifs+funny+me_irl

You should write them as plain and put plus between them. Don't use curly
braces.

**Comment by aliparlakci** [comment]
I think I understood your problem. What cat does is basically executing the script for each subreddit on the list one by one, waiting one to finish before executing another.

**There is a `--quit` argument for this type of problem. You can add it at the end of your command and it should auto exit when the job finishes.** 

The reason script doesn't automatically quits is to be able to see the error codes when run via double clicking on the executable.

**Comment by nihelmasell** [comment]
> I think I understood your problem. What cat does is basically executing the script for each subreddit on the list one by one, waiting one to finish before executing another.
> 
> **There is a `--quit` argument for this type of problem. You can add it at the end of your command and it should auto exit when the job finishes.**
> 
> The reason script doesn't automatically quits is to be able to see the error codes when run via double clicking on the executable.
> 
> Before you go, can tell me how professional you find the program and is there any features you don't like? You seem to be a serious user.

Hi, indeed, I started collecting lots of stuff from the Internet like 10 years ago. Before of this I did it thru the newsgroups, but now they are almost dead, as much more interesting stuff is found on Reddit and other some social media. The program is pretty stable, and has the advantages over Ripme (the other script I use) that a) it can save comments  b) it labels files correctly -with the post name- and c) it makes subdirectories when a user posts an album (Ripme started doing this, but didn't before). The only issues I find are: a) It is impossible it for it to stand 900 subreddits via cat (maybe it is the terminal memory, though) and b) that you cant run it with all sorting types (that is, hot, new, controversial, etc) at the same time -I think thats what Ripme does- so you have to write six lines into the script, repeating all of these sorting options.

Regarding the cat command, I m going to try your suggestion. Are you totally sure that's also causing the repeating login attempts? It pretty weird

Best Regards, and great app!

**Comment by aliparlakci** [comment]
You didn't mention about the recurring login attempts at your *first* message. Actually, I can't think of any reason to it? You should provide me with the CONSOLE_LOG.txt file. It logs the terminal outputs. And also your login info (not your username or password) are held at your home path inside the config.cfg file. Can you check there if your info is still there or deleted somehow. Alternatively, you can copy and paste the config.cfg file with your login info next to the script.

**Comment by nihelmasell** [comment]
(Sorry I can’t quote, I’m on mobile). Can you summarize the advantages of the newest version over the jul 22 version? It seems to support the same list of sites. Does it have more threads? I will send you the log file into my next response (I promise). I’m using that version, which doesn’t have that login issue. And for the record, I’m on a Mac, with high Sierra. That version allowed to have my username and password into the config file. Maybe the problem is related to that.

**Comment by aliparlakci** [comment]
Not much have changed since July 22 as for features. Couple of sites are now supported. But the reason I suggest you to use the latest version is I have fixed loads of bugs since then. I don't remember most of them. Also, older version signs into reddit using your password which is stored in a plain text file without any encryption. This is not secure at all. You can still use the older version but you might encounter bugs.

If you want me to find why script is constantly relogins to reddit, you should give me some technical details and tell how exactly it happens.

**Comment by nihelmasell** [comment]
> Not much have changed since July 22 as for features. Couple of sites are now supported. But the reason I suggest you to use the latest version is I have fixed loads of bugs since then. I don't remember most of them. Also, older version signs into reddit using your password which is stored in a plain text file without any encryption. This is not secure at all. You can still use the older version but you might encounter bugs.
> 
> If you want me to find why script is constantly relogins to reddit, you should give me some technical details and tell how exactly it happens.

Hi, I searched the logs. This is what the log for each subreddit ran by cat says:

"script.py --directory /Volumes/WD-8TB-NOV2018/bulk-downloader-for-reddit-master/Backup --sort hot --subreddit funny 

Bulk Downloader for Reddit v1.6.4

Go to this URL and login to reddit:

 https://www.reddit.com/api/v1/authorize?client_id=BSyphDdxYZAgVQ&duration=permanent&redirect_uri=http%3A%2F%2Flocalhost%3A1337&response_type=code&scope=identity+history+read&state=13907"

Each time the script finishes with a subreddit fed by cat, you have to authorize the app again. It downloads all files ok, the problem is you have to renew the authorization each time cat feeds a subreddit into the app. I tried editing manually the config file, but the script asks for permission again. I also tried the config file from the old version (the one Im running now) and also tried deleting the config file, forcing the app to make a new one. IT ALWAYS ASKS for permission AGAIN. For each subreddit. Maybe is something related to my OS. I also tried with python2 and 3. But to fully emulate the bug, you should have a MacOS virtual machine. I suppose its something related to my OS.

**Comment by aliparlakci** [comment]
Can you check if the script creates a config file and fills them with credentials?

If it does, for a quick fix, move that file **next to** the script. It should do the trick.

**Comment by aliparlakci** [comment]
Can you check the latest commit on the master branch? I might have fixed the problem. Sorry for the deal, btw. I have been very busy lately.

**Comment by nihelmasell** [comment]
Hi, no problem. I will just check it as soon as I finish the latest update (I have like 1100 subreddits now and run the script with all the sort types -6 times for all subs-). PS I notice it works fine with parallel 5 threads! That’s great!

**Comment by nihelmasell** [comment]
> Can you check the latest commit on the master branch? I might have fixed the problem. Sorry for the deal, btw. I have been very busy lately.

Hi, thanks a lot! It seems to be working now!!
